src/section/shared_state.rs

```diff
@@ -40,9 +40,9 @@ pub(crate) struct SharedState {
 }
 
 impl SharedState {
-    pub fn new(elders_info: Proven<EldersInfo>) -> Self {
+    pub fn new(section_key: bls::PublicKey, elders_info: Proven<EldersInfo>) -> Self {
         Self {
-            our_history: SectionProofChain::new(elders_info.proof.public_key),
```

Comment:
just want to confirm that `elders_info.proof.public_key` is the DKG key with participants of `elders_info.value (i.e. the elders)`? i.e. the section_key of the elders of the `elders_info`, right ?

Reply:
No, that's the issue this PR is trying to fix. The key the `elders_info` is signed with is not the latest section key - it's the one before that. The reason for this is that we vote for the elders_info and the new section key at the same time, so we don't yet have the new key in the section chain. We could probably change it so that the elders_info is signed with the latest key (because we do have the new DKG result at that time), and maybe we should do it, but that's not how things currently work.

Reply:
Ok, I got it.
Thx for the clarification

Reply:
maybe worth to put your above wording as a comment in the code ?

Reply:
:+1:  will do.